app/UserPreferences.m

```diff
@@ -166,29 +182,43 @@ - (BOOL)isEqual:(id)object {
 
 NSDictionary<NSString *, Theme *> *presetThemes;
 + (void)initialize {
-    presetThemes = @{@"Light": [self _themeWithForegroundColor:UIColor.blackColor
-                                               backgroundColor:UIColor.whiteColor],
-                     @"Dark":  [self _themeWithForegroundColor:UIColor.whiteColor
-                                               backgroundColor:UIColor.blackColor],
-                     @"1337":  [self _themeWithForegroundColor:UIColor.greenColor
-                                               backgroundColor:UIColor.blackColor]};
+    presetThemes = @{@"System": [self _themeWithForegroundColor:[UIColor colorNamed:@"systemForegroundColor"]
+                                                backgroundColor:[UIColor colorNamed:@"systemBackgroundColor"]
+                                                     identifier:@0],
+                     @"Light":  [self _themeWithForegroundColor:UIColor.blackColor
+                                                backgroundColor:UIColor.whiteColor
+                                                     identifier:@1],
+                     @"Dark":   [self _themeWithForegroundColor:UIColor.whiteColor
+                                                backgroundColor:UIColor.blackColor
+                                                     identifier:@2],
+                     @"1337":   [self _themeWithForegroundColor:UIColor.greenColor
+                                                backgroundColor:UIColor.blackColor
+                                                     identifier:@3]};
 }
 
 + (NSArray<NSString *> *)presetNames {
-    return @[@"Light", @"Dark", @"1337"];
+    NSArray* names = @[];
+    
+    // Only iOS versions with dark mode have an Auto theme
+    if(SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"13.0"))
+        names = [names arrayByAddingObjectsFromArray:@[@"System"]];
+    
+    return [names arrayByAddingObjectsFromArray:@[@"Light", @"Dark", @"1337"]];
 }
 + (instancetype)presetThemeNamed:(NSString *)name {
     return presetThemes[name];
 }
 - (NSString *)presetName {
     for (NSString *name in presetThemes) {
-        if ([self isEqual:presetThemes[name]])
+        if ([self isEqual:presetThemes[name]]) {
```

Comment:
Could you leave this formatting alone? You're not changing this part of the file, so this commit would show up unnecessarily in git blame.

Reply:
Done